app/assets/javascripts/components/student/student_comments.es6.jsx

```diff
@@ -37,7 +49,17 @@ class StudentComments extends Component {
   }
 
   renderStudentComments() {
-    return this.props.comments.map(this.renderStudentComment.bind(this));
+    return this.state.comments.map((comment, index) => this.renderStudentComment(comment, index));
+  }
+
+  createCommentRequest(params) {
+    resolve = (response) => this.setState({ comments: response });
```

Comment:
This `resolve` callback doesn't seem right - if my guess is right you'll find out why when you test it out.